platform/windows/detect.py

```diff
@@ -328,7 +346,7 @@ def configure_mingw(env):
         env.Append(CCFLAGS=["-msse2"])
 
         if env["optimize"] == "speed":  # optimize for speed (default)
-            if env["bits"] == "64":
+            if env["arch"] == "x86_64":
                 env.Append(CCFLAGS=["-O3"])
             else:
                 env.Append(CCFLAGS=["-O2"])
```

Comment:
That's probably were added specifically for `x86`, and should be:

```
            if env["arch"] == "x86_32":
                env.Append(CCFLAGS=["-O2"])
            else:
                env.Append(CCFLAGS=["-O3"])
```

Reply:
Do we know why `x86_32` would need to use `-O2` instead of `-O3`?
Can there be similar issues on `arm32`?